jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -104,20 +108,40 @@ public boolean enqueue(Frame frame, Callback callback, boolean batch)
                 dead = closedCause;
                 if (dead == null)
                 {
-                    if (opCode == OpCode.PING || opCode == OpCode.PONG)
-                        queue.offerFirst(entry);
-                    else
-                        queue.offerLast(entry);
+                    switch (opCode)
+                    {
+                        case OpCode.CLOSE:
+                            closeStatus = CloseStatus.getCloseStatus(frame);
+                            if (CloseStatus.isOrdinary(closeStatus))
```

Comment:
Make sure we are "ordinary" also for close codes >= 3000 (see https://tools.ietf.org/html/rfc6455#section-7.4.2).

Reply:
currently we have classified `SHUTDOWN` 1001 as an ordinary status code. However now I am thinking that this should be abnormal as well.

Reply:
I agree SHUTDOWN should be abnormal.   The server is stopping, so no point continuing to send your 4GB message!